src/main/java/KilimanJARo/P2P/client/ClientServerController.java

```diff
@@ -41,6 +51,71 @@ public ClientServerController(@Qualifier("ClientServerRestTemplate") RestTemplat
+        // headers.setBasicAuth("username", "password");
+        // headers.set("Content-Type", "application/json");
+        RequestEntity<RegisterRequest> requestEntity =
+            RequestEntity.post(URI.create(central_properties.getProperty("server.url.apu.register")))
+            .headers(headers)
+            .body(request);
+
+        ResponseEntity<RegisterResponse> response = restTemplate.exchange(requestEntity, RegisterResponse.class);
+        password = response.getBody().getPassword();
+
+        if (response.getBody() != null && response.getBody().isSuccess()) {
+            return ResponseEntity.ok(new RegisterResponse(true, "Server registered successfully", response.getBody().getPassword()));
+        } else {
+            return ResponseEntity.status(500).body(new RegisterResponse(false, "Server registration failed", null));
+        }
+    }
+
+    @GetMapping("/authWithMainServer")
+    public ResponseEntity<AuthResponse> authWithMainServer(@RequestParam String password) {
+        AuthRequest request = new AuthRequest("ClientServer", password);
+        HttpHeaders headers = new HttpHeaders();
+        // headers.setBasicAuth("username", "password");
+        // headers.set("Content-Type", "application/json");
+
+        RequestEntity<AuthRequest> requestEntity = RequestEntity
+                .post(URI.create(central_properties.getProperty("server.url.api.login")))
+                .headers(headers)
+                .body(request);
+
+        ResponseEntity<AuthResponse> response = restTemplate.exchange(requestEntity, AuthResponse.class);
+        password = response.getBody().getNextPassword();
+
+        if (response.getBody() != null && response.getBody().isSuccess()) {
+            return ResponseEntity.ok(new AuthResponse(true, "Server authenticated successfully", response.getBody().getNextPassword()));
+        } else {
+            return ResponseEntity.status(500).body(new AuthResponse(false, "Server authentication failed", null));
+        }
+    }
+
+    @GetMapping("/logoutFromMainServer")
+    public ResponseEntity<LogoutResponse> logoutFromMainServer() {
+        LogoutRequest request = new LogoutRequest("ClientServer");
+        HttpHeaders headers = new HttpHeaders();
+        // headers.setBasicAuth("username", "password");
+        // headers.set("Content-Type", "application/json");
+
+        RequestEntity<LogoutRequest> requestEntity = RequestEntity
+                .post(URI.create(central_properties.getProperty("server.url.api.logout")))
+                .headers(headers)
+                .body(request);
+
+        ResponseEntity<LogoutResponse> response = restTemplate.exchange(requestEntity, LogoutResponse.class);
+
+        if (response.getBody() != null && response.getBody().isSuccess()) {
+            return ResponseEntity.ok(new LogoutResponse(true, "Logged out successfully"));
+        } else {
+            return ResponseEntity.status(500).body(new LogoutResponse(false, "Logout failed"));
```

Comment:
Looking like we need some more detailed response for logout error. (May-be add issue?)